addons/hr_holidays/models/hr_leave_allocation.py

```diff
@@ -150,6 +150,13 @@ def _holiday_status_id_domain(self):
         ('interval_number_check', "CHECK(interval_number > 0)", "The interval number should be greater than 0"),
     ]
 
+    @api.constrains('holiday_status_id')
+    def _check_allocation_type(self):
+        for rec in self:
+            if rec.env.context.get('force_allocation_type', []) and rec.holiday_status_id.allocation_type == "no":
+                raise ValidationError(
+                    "This time of type \"{0}\" does not require allocation because \"No Allocation Needed\" set in Allocation mode.".format(self.name))
```

Comment:
This term untranslated. You should use ``_()`` (and add the term in the .pot file as you are working on a stable release ;)

Reply:
Better wording proposal: 
```This time off type doesn't require allocations.```